### DISCUSSION TITLE
Why extracting boundary points but got all the vertices returned?

### DISCUSSION THREAD

**Discussion by 1939938853** [pyvista/pyvista#5012] [Q&A]
Hello,

I'm trying to extract the boundary points, but got all the vertices returned? What can be the problem? Below is my code and the result screenshot.

Than you!

`
import pyvista as pv
boundary = mesh.extract_feature_edges(boundary_edges=True, 
                                      non_manifold_edges=False, 
                                      manifold_edges=False)

p = pv.Plotter()
p.add_mesh(surf, color='tan', show_edges=True)
p.add_points(boundary.copy(), color='red', point_size=10)
p.show()
`

![image](https://github.com/pyvista/pyvista/assets/59957922/8d9402dd-7fb9-4728-9e11-8a0d853e399a)

**Comment by darikg** [comment]
Perhaps the mesh is a triangle soup? If so, a `mesh = mesh.clean()` beforehand should fix it. Otherwise, could you share the mesh itself?

To be careful you'll probably want to add `feature_edges=False` to that `extract_feature_edges` call but I don't think that's causing your problem.
